bindings/python/test/test_builders.py

```diff
@@ -54,33 +54,40 @@ def test_default_unit(self):
         builder = DatetimeBuilder()
         self.assertEqual(builder.unit, timestamp('ms'))
 
-    def _test_simple(self, tstamp_units, kwarg_name):
-        builder = DatetimeBuilder(dtype=timestamp(tstamp_units))
-        datetimes = [datetime(1970, 1, 1) + timedelta(**{kwarg_name: k*100})
+    def _to_int(self, dt):
+        # datetime.timestamp assumes local time so we must explicitly
+        # specify tzinfo=timezone.utc to obtain POSIX time
+        return int(dt.replace(tzinfo=timezone.utc).timestamp() * 1000)
```

Comment:
Is this timezone stuff still relevant given we use `datetime.utcnow` in the tests?

Reply:
Yeah, unfortunately. the `.timestamp()` method always assumes times to be local. See the docs - https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp

Reply:
FYI this is what we do in pymongo:
```python
def _datetime_to_millis(dtm):
    """Convert datetime to milliseconds since epoch UTC."""
    if dtm.utcoffset() is not None:
        dtm = dtm - dtm.utcoffset()
    return int(calendar.timegm(dtm.timetuple()) * 1000 +
               dtm.microsecond // 1000)
```
https://github.com/mongodb/mongo-python-driver/blob/3.11.3/bson/__init__.py#L902-L907

Reply:
I think there is a misunderstanding here. The Python docs say:

> **Naive datetime** instances are assumed to represent local time...
> For **aware datetime** instances, the return value is computed as:
>    (dt - datetime(1970, 1, 1, tzinfo=timezone.utc)).total_seconds()

As it stands the current `_to_int` doesn't work correctly for aware datetimes. `replace(tzinfo=timezone.utc)` truncates the original timezone and replaces it with UTC but doesn't apply the offset. We should use pymongo's method which we know is correct.

Reply:
Vendored that method from `bson` into the test class.